src/ImageSharp/Formats/Jpeg/Components/Encoder/HuffmanScanEncoder.cs

```diff
@@ -246,9 +246,9 @@ public void EncodeScanBaseline(Component component, CancellationToken cancellati
     private void EncodeScanBaselineInterleaved<TPixel>(JpegFrame frame, SpectralConverter<TPixel> converter, CancellationToken cancellationToken)
         where TPixel : unmanaged, IPixel<TPixel>
     {
-        nint mcu = 0;
-        nint mcusPerColumn = frame.McusPerColumn;
-        nint mcusPerLine = frame.McusPerLine;
+        int mcu = 0;
```

Comment:
`mcu` is only used as `int`. 

Places where the variable is used for addressing (i.e. memory reads and writes) native-integers are used (preferable `nuint`), so that no sign-extending moves happen, as a 32-bit `int` has to be sign-extended on 64-bit to do the memory operation.